server/command_hooks.go

```diff
@@ -70,6 +70,16 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
 		}
 	}
 
+	conf := p.getConfiguration()
+	if conf.EnableAdminRestrictions {
```

Comment:
We can for the API, but in the command hook I am not sure if it makes sense because like you said the user already needs to be part of the channel in order to invoke the command. Also it seems to make more sense sequentially to check if the user even has the correct permissions before we call `p.hasPermissionToChannel` as that has to retrieve the channel in its current implementation, afterwards we then again retrieve the channel in command hook, so that logic seems like it would just introduce a slight performance hit and redundant checks (which will always return true). (https://github.com/mattermost/mattermost-plugin-channel-export/blob/32244afbecfd70d3ff42e3fc61c03a9e93a495b8/server/command_hooks.go#L83)

It may be worth it in terms of simplicity of code but that call is up to the maintainers.

Reply:
Would it makes sense to refactor the new logic into one function that is called from both entrypoints?

Reply:
Yeah, I think that makes the most sense given the duplicate functionality.

Reply:
Updated to have a method to remove duplicate code
https://github.com/mattermost/mattermost-plugin-channel-export/blob/11a52c9b4235d56cf5a754cfa49e9e69f0562b13/server/command_hooks.go#L196